Awsamplify case study (#176)
NoroffFEU/development-platforms: src/content/docs/baas/AWS Amplify/awsamplify-case-study.md

---
title: AWS Amplify
keywords: Backend as a service (BaaS), AWS Amplify, File Storage, Authentication, Real-time database
tags: Backend as a service (BaaS), AWS Amplify, File Storage, Authentication, Real-time database
---

## Introduction

> AWS Amplify from Amazon is a tool that helps developers create secure and scalable apps more easy. It takes care of many technical aspects like user authentication, real-time data updates, and server functions, so developers can focus on making app features without worrying about difficult technical details.

Reviewer: ```suggestion
AWS Amplify from Amazon is a tool that helps developers create secure and scalable apps more easily. It takes care of many technical aspects like user authentication, real-time data updates, and server functions, so developers can focus on making app features without worrying about difficult technical details.
```

## Brief History

AWS Amplify was introduced in late 2017 and was emerged as a specialized platform within AWS (Amazon Web services). It aimed to streamline web and mobile applications development, offering services that made managing backend infrastructure a bit easier.

> The origins of Amplify lie in AWS's goal to simplify app and web development. With increased demand for good and scalable applications, AWS made amplify to offer tools to developers for an easier and quicker process. Amplify addressed this by providing a lot of services, including authentication, database management, real-time data synchronization, analytics and more. This enabled developers to focus more on the frontend and on core functions on their apps.

Reviewer: ```suggestion
The origins of Amplify lie in AWS's goal to simplify app and web development. With increased demand for good and scalable applications, AWS made Amplify to offer tools to developers for an easier and quicker process. Amplify addressed this by providing a lot of services, including authentication, database management, real-time data synchronization, analytics and more. This enabled developers to focus more on the frontend and on core functions in their apps.
```

> Since it was made, Amplify has continued to evolve and has expanded. It has incorporated new features and services, such as better integrations with other AWS services, better security measures, and more tools for app deployment and managaement.

Reviewer: ```suggestion
Since it was made, Amplify has continued to evolve and has expanded. It has incorporated new features and services, such as better integrations with other AWS services, better security measures, and more tools for app deployment and management.
```

## Core Features

> - **Authentication:** Amplify provides built in authentication services, such as user sign up, sign in, and control to secure app data

Reviewer: ```suggestion
- **Authentication:** Amplify provides built-in authentication services, such as user sign up, sign in, and control to secure app data
```

It's unclear what "control to secure app data" means.

> - **APIS and GraphQL:** Offers tools to create and manage APIS, supporting REST, GraphQL and WebSocket protocols for better data interaction

Reviewer: ```suggestion
- **APIS and GraphQL:** Offers tools to create and manage APIs, supporting REST, GraphQL and WebSocket protocols for better data interaction
```

- **Real-Time Data Synchronization:** Enables real-time updates and synchronization of data across clients, ensuring fast and consistent user experiences.

> - **Serverless Funcitons:** Can deploy serverless fucntions, allowing developers to make code without managing difficult infrastructure

Reviewer: ```suggestion
- **Serverless Funcitons:** Can deploy serverless functions, allowing developers to write code without managing difficult infrastructure
```

- **File Management:** Has secure and scalable file storage capabilities for managing user-generated content.

> - **Analytics:** Offers tools to see insight on app usage, user behavior, performance and app-optimization.

Reviewer: ```suggestion
- **Analytics:** Offers tools to see insights on app usage, user behavior, performance and app-optimization.
```

- **Hosting and Deployment:** Has a simple hosting and deployment process supporting automated deployment of web applications.

## Strengths
_Several core features is what amplify has as its strength_

- **Simple Backend Development:** Amplify makes backend tasks simple, enabling developers to focus more on frontend and core app functionalities
- **User Authentication:** Has robust and easy built in authentication services for secure user management

> - **Real-Time Data Synchronization:** has real-time data updates across various app clients for better and consistent experiences.

Reviewer: ```suggestion
- **Real-Time Data Synchronization:** Has real-time data updates across various app clients for better and consistent experiences.
```

- **Scalability:** Provides scalable infrastructure, which means apps can grow without massive architectural changes

> - **Integrated Analytics:** has tools for app usage, user behavior insights and can aid in app optimization.

Reviewer: ```suggestion
- **Integrated Analytics:** Has tools for app usage, user behavior insights and can aid in app optimization.
```

## Weaknesses

- **Limited Customization:** May have limitiations in customization for users that want specific customization.
- **Complex Configuration:** Difficult configurations demands knowledge or deeper understanding of AWS and their services
- **Dependent on AWS Ecosystem:** Deep integration limits flexibility for people with multi-cloud preferences or other non-AWS infrastructure
- **Cost and Price:** has complex cost structure, which can be challenging during app scaling
- **Learning Curve:** Has a learning curve for initial setup, can be challenging for some new users.

## Comparison - AWS Amplify vs Firebase
1. **Ecosystem:**
   - AWS Amplify: Provided by Amazon Web Services, is deeply integrated with the AWS ecosystem. Its strength is extensive scalability, and flexible infrastructure.
   - Firebase: Provided by Google, with strong integration with Google Cloud but not as vast as the AWS ecosystem. Is known for its simplicity and ease of use

2. **Scalability and Flexibility:**
   - AWS Amplify: Amplify has very good scalability, which is ideal for enterprise-level projects or for those who are seeking more customization for the backend infrastructure.

>    - Firebase: Firebase offers a more userfriendly and scalable platform, is better for smaller projects, startups, and developers who are looking for rapid integration with simpler configuration requirements.

Reviewer: ```suggestion
   - Firebase: Firebase offers a more user-friendly and scalable platform, is better for smaller projects, startups, and developers who are looking for rapid integration with simpler configuration requirements.
```

3. **Pricing and Cost Structure:**
   - AWS Amplify: Amplify has a more complicated and usage-dependent pricing model.

>    - Firebase: Offers a more clear pricing model. including a free tier, has straight forward payment plans based on usage, which makes it better for smaller projects.

Reviewer: ```suggestion
   - Firebase: Offers a more clear pricing model, including a free tier. Has straight-forward payment plans based on usage, which makes it better for smaller projects.
```

  
4. **Development Flexibility and Learning Curve:**
   - AWS Amplify: Has very good customization and flexibility options, which might lead to a steeper learning curve, especially for newcomers to AWS or backend services

>    - Firebase: Is known for its userfriendly interface and simplified setup. Firebase also has a shorter learning curve than Amplify, making it better for beginners.

Reviewer: ```suggestion
   - Firebase: Is known for its user-friendly interface and simplified setup. Firebase also has a shorter learning curve than Amplify, making it better for beginners.
```
